source/blog/2017-05-12-asynchronous-python-sdk.html.md

```diff
@@ -0,0 +1,127 @@
+With HTTP Pipelining client can send multiple requests without waiting for ther server response. Only idempotent HTTP methods can be pipelined.
+
+```
+With pipelining disabled            With pipelining enabled
+
+       |     |                            |      |
+       | \   |                            | \    |
+       |  \  |                            |  \   |
+       |   \ |                            | \ \  |
+       |    \|                            |  \ \ |
+       |    /|                            |   \ \|
+       |   / |                            |    \/| 
+       |  /  |                            |    /\|
+CLIENT | /   | SERVER              CLIENT |   / /| SERVER
+       | \   |                            |  / / |
+       |  \  |                            | / /  |
+       |   \ |                            |/ /   |
+       |    \|                            | /    |
+       |    /|                            |      |
+       |   / |                            
+       |  /  |                            
+       | /   |                            
+```
+
+## SDK implementation
+
+To not break the backward compatibility of the SDK, we have introduced a new boolean parameter called ```wait```, to the methods of SDK services.
+The default values is ```True```, so it's working in synchronous fashion. If the user send the ```wait=False```, the SDK will send a request specified by user program and will return the ```Future``` object,
+which is defined as follows:
+
+```python
+class Future(object):
+    """
+    Instances of this class are returned for operations that specify the
+    `wait=False` parameter.
+    """
+
+    def wait(self):
+        """
+        Waits till the result of the operation that created this future is
+        available.
+        """
+```
+
+So in order to wait for the response user need to call the ```wait``` method, which will return the resultion object.
+
+## Example
+
+```python
+import ovirtsdk4 as sdk
+
+connection = sdk.Connection(
+    url='https://example.engine.com/ovirt-engine/api',
+    username='admin@internal',
+    password='123456',
+    ca_files='ca.pem',
+    connections=5,
+    pipeline=5,
+)
+
+system_service = connection.system_service()
+
+vms_service = system_service.vms_service()
+vms = system_service.vms_service().list()
+
+futures = {}
+for vm in vms:
+    vm_service = vms_service.vm_service(vm.id)
+    disks_future = vm_service.disk_attachments_service().list(wait=False)
+    nics_future = vm_service.nics_service().list(wait=False)
+    perms_future = vm_service.permissions_service().list(wait=False)
+    futures[vm.name] = [disks_future, nics_future, perms_future]
+
+for vm_name, vm_resources in futures.iteritems():
+    for resource in vm_resources:
+        resource.wait()
+
+connection.close()
+```
+
+You can check another [example] in git repository of Python SDK.
```

Comment:
in git repository of Python SDK -> in the git repository of the oVirt Python SDK, under the examples sub directory.